### PR TITLE
Tzwong/keras agent

### PR DESCRIPTION
**Pull request by wes1350**

### CODE REVIEW

File: Config.py [hunk 1]
```diff
@@ -5,11 +5,15 @@
 
 class Config:
     """The class storing all the config parameters."""
-    def __init__(self, **kwargs) -> None:
+    def __init__(self, **kwargs) -> None: 
         self.local_ais = {}
+        # initialize other parameters
+        for key, value in kwargs.items():
```
Comment (wes1350) [inline Config.py:11]: Why did you move this first? Now some specified settings in kwargs will be overwritten by what is below right?

File: agents/utils/network.py [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+def unimplemented_update():
+    def do_nothing(event):
+        pass
+    return do_nothing
+
+def start(on_action=unimplemented_response("actions"), 
+          on_reaction=unimplemented_response("reactions"), 
+          on_card=unimplemented_response("card selection"), 
+          on_exchange=unimplemented_response("exchanges"),
+          update_f=unimplemented_update()):
```
Comment (wes1350) [inline agents/utils/network.py:20]: All of this is in the Agent class now - why was it added back?

File: agents/utils/network.py [hunk 3]
```diff
@@ -14,34 +27,44 @@ def on_prompt(message):
...
+        else:
+            if not isinstance(event_type, str):
+                raise ValueError("event_type must be a string type")
+            else:
+                raise ValueError("Received unknown event type: " + event_type)
+        if response is None:
+            raise Exception(("Agent did not return a value for its response. "
+                             "Make sure to return a value (e.g. return income()) when choosing a response."))
+        return response
+
```
Comment (wes1350) [inline agents/utils/network.py:52]: Also confused why this was added back

File: Config.py (hunk 1 above)
Comment (tzerchyuan) [reply Config.py:11]: I was thinking about something else that does not matter now, will revert

File: agents/utils/network.py (hunk 2 above)
Comment (tzerchyuan) [reply agents/utils/network.py:20]: I think it is cause I didn't have the changes

File: agents/utils/network.py (hunk 3 above)
Comment (tzerchyuan) [reply agents/utils/network.py:52]: it has to do with not getting the right changes, none of the reverting is intentional
